### PR TITLE
Rename enc service to storage

### PR DESCRIPTION
**Pull request by matas-cc**
### Description
Renaming enc/encryption service to storage service, as a preparation for introducing the encryption service, which does not use storage.

### Parent Issue
Main:
https://github.com/cyber-crypt-com/encryptonize-core/issues/192

Subtask:
https://github.com/cyber-crypt-com/encryptonize-core/issues/193

### Comments for the Reviewers

Changes in core, ECCS. I'll make a separate PR for premium.

To sum up, changes are:

Package naming
enc -> storage

Main service struct
Enc -> Storage

Changed the receiver naming on service
(enc *Enc) -> (strg *Storage)

gRPC path:
/enc.Encryptonize/ -> /storage.Encryptonize/

### Type(s) of Change (Split the PR if you check many)
- [ ] Added user feature
- [ ] Added technical feature
- [ ] Added tests
- [x] Refactor
- [ ] Bugfix
- [ ] Updated documentation
- [ ] Infrastructure changes (Terraform, GCP, K8s, other)
- [ ] CI/CD workflow changes

### Testing Checklist
- [ ] I have added/updated unit tests for functional changes.
- [ ] I have added/updated integration tests for changes that affect dependent systems.
- [ ] I have added/updated end-to-end tests for feature changes.

### General Checklist
- [x] I have pulled in the latest changes from master.
- [x] I have run `make lint`.
- [x] I have successfully run `make docker-up; make tests`.
- [ ] I have updated relevant workflows and deployment tools.
- [ ] I have updated/added relevant documentation (readme, doc comments, etc).
- [ ] I have added the license to new source code files.
- [x] I have spent some time looking over the full diff before creating this PR.

### CODE REVIEW

**Comment by matas-cc** [comment]
Targeted dependency layout after refactoring and adding new service

image

**Review by philip-cc — COMMENTED** [review]
This looks good, but I think we need to update `api-v2.md` as well. Actually, it's even outdated currently, as the split into `app`, `enc`, `auth` isn't even reflected. Please have a look!

File: documentation/api/api-v2.md [hunk 1]
```diff
@@ -207,7 +213,7 @@ running encryptonize deployment.
 
 # Store
 
-Takes an `Object` and Stores it in encrypted form. This call can fail if the Encryption Service
+Takes an `Object` and Stores it in encrypted form. This call can fail if the Storage Service
```
Comment (philip-cc) [inline documentation/api/api-v2.md:216]: I think we should be precise and use `storage.Object`, `storage.Store`, etc.

Comment (matas-cc) [reply documentation/api/api-v2.md]: Updated

**Review by philip-cc — APPROVED** [review]
Ship it!

**Pull request merged**